src/components/pages/profile/account/SetMFA.vue

```diff
@@ -0,0 +1,371 @@
+<!--
+  # SetMFA
+  This component is used to set up MFA for a contact.
+  It uses a slider to guide the user through the process.
+
+  ## Props
+  - `contactId` (string): The id of the contact to set up MFA for.
+
+  ## Possible improvements
+  - Add support for other MFA types (e.g. SMS)
+  - Transform this component into a general useable wizard component
+
+-->
+
+<template>
+  <AppHeading class="my-3">
+    {{ t('accountPage.mfa.title') }}
+  </AppHeading>
+
+  <AppButton
+    v-if="isEnabled"
+    variant="primaryOutlined"
+    :icon="faMobileAlt"
+    @click="disableMFA"
+  >
+    {{ t(`accountPage.mfa.disable`) }}
+  </AppButton>
+
+  <AppButton
+    v-else
+    variant="primaryOutlined"
+    :icon="faMobileAlt"
+    @click="toggleMFAModal"
+  >
+    {{ t(`accountPage.mfa.enable`) }}
+  </AppButton>
+
+  <AppModal
+    :open="showMFASettingsModal"
+    :title="modalTitle"
+    class="w-full"
+    @close="onCloseMFAModal"
+  >
+    <AppStepper
+      v-model="appStepper.selectedStepIndex"
+      :steps="appStepper.steps"
+      @update:model-value="onStepperChange"
+    />
+    <AppSlider ref="appSliderCo" @slide="onSlideChange">
+      <template #slides>
+        <AppSlide>
+          <div class="whitespace-break-spaces">
+            <p class="text-center">
+              {{ t(`accountPage.mfa.scan.desc`) }}
+            </p>
+            <AppQRCode v-if="totpUrl" :qr-data="totpUrl" />
+            <!-- TODO: Allow other options like show url / secret key here? -->
```

Comment:
As a rule you should always give the option to copy the secret key directly I think, for those who want to add the code to managed service on their computer like LastPass. Can you just add it as a quick readonly input?